packages/devtools_app/lib/src/config_specific/flutter/import_export/import_export.dart

```diff
@@ -13,12 +17,16 @@ import '_fake_export.dart'
 // TODO(kenz): we should support a file picker import for desktop.
 
 class ImportController {
-  const ImportController(this._notifications, this._controllers);
+  ImportController(this.scaffoldState) {
```

Comment:
This is highly unusual.

This class should describe its expected lifecycle in detail to make sure we're not throwing out a scaffold state that this depends on, as well as making sure this class is properly able to respond to the Notifications and Controllers objects it depends on changing.

Reply:
Simplified to just pass in notifications, controllers, and a callback to the ImportController. This is simpler and much better. Good suggestion!